### PR TITLE
Feature/merging develop ver.30

### PR DESCRIPTION
**Pull request by dk1844**
This is essentially merging `develop` into `develop-ver3.0`, but since it was non-trivial, this PR was created.

### CODE REVIEW

**Comment by benedeki** [comment]
Non-trivial 😞 
Then I would wait the develop to be updated to the expected (hopefully) version - e.g. merge of #1740

File: menas-web/Dockerfile [hunk 1]
```diff
@@ -11,7 +11,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM nginx:1-alpine
+ARG DOCKER_BASE_ARTIFACT=nginx
+ARG DOCKER_BASE_TAG=1-alpine
+
+FROM ${DOCKER_BASE_ARTIFACT}:${DOCKER_BASE_TAG}
```
Comment (Zejnilovic) [inline menas-web/Dockerfile:17]: I know this is a new feature that people are excited about, but I think it was mainly created for multi staged builds.

What does it give us here? What are the pros of this approach? I honestly see only possible problems here + the extra (undeclared) properties in Maven.

File: menas/src/main/resources/docker.properties.template [hunk 2]
```diff
@@ -34,11 +34,18 @@ spring.resources.chain.strategy.fixed.version=@project.version@
...
+menas.auth.jwt.lifespan.hours=8
+
 # Timeout (in seconds) for MongoDB migration queries
 migrations.mongo.query.timeout.seconds=300
 
 menas.spark.master=local[1]
 
+# The template for Lineage (Spline) display. If empty Lineage button is not shown.
+# Avoiding the http(s) protocol will cause that the same protocol will be used as for Menas itself, avoiding issues
+spline.urlTemplate=//localhost:8080/spline/dataset/lineage/_search?path=%s&application_id=%s
```
[the remark this answers is not in the document]
Comment (AdrianOlosutean) [reply menas/src/main/resources/docker.properties.template:47]: I think this is not even needed anymore, since it will be removed from spline integration

**Comment by dk1844** [comment]
#1740 merged into `develop`, updated this PR/branch by merging `develop` into it.

File: dao/src/main/scala/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -1,3 +1,3 @@
-Manifest-Version: 1.0
-Class-Path: 
-
+Manifest-Version: 1.0
```
Comment (AdrianOlosutean) [inline dao/src/main/scala/META-INF/MANIFEST.MF:1]: Let's remove this file

**Review by AdrianOlosutean — DISMISSED** [review]

File: dao/src/main/scala/META-INF/MANIFEST.MF (hunk 3 above)
Comment (dk1844) [reply dao/src/main/scala/META-INF/MANIFEST.MF:1]: This is just a CRLF->LF switch, a possible removal can be done separately.

File: menas-web/Dockerfile (hunk 1 above)
Comment (Zejnilovic) [reply menas-web/Dockerfile:17]: I would put a comment in there above the ARGVs as to why it is like that.

File: menas/src/main/resources/docker.properties.template (hunk 2 above)
[the remark this answers is not in the document]
Comment (dk1844) [reply menas/src/main/resources/docker.properties.template]: @dvagapov removed it.

**Review by Zejnilovic — APPROVED** [review]
I only went over Docker stuff

**Review by AdrianOlosutean — APPROVED** [review]

**Pull request merged**